framework/src/node/generator/endpoint.ts

```diff
@@ -0,0 +1,229 @@
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+import { Chain } from '@liskhq/lisk-chain';
+import { StateStore } from '@liskhq/lisk-chain/dist-node/state_store';
+import {
+	decryptPassphraseWithPassword,
+	getAddressFromPublicKey,
+	getPrivateAndPublicKeyFromPassphrase,
+	parseEncryptedPassphrase,
+} from '@liskhq/lisk-cryptography';
+import { KVStore } from '@liskhq/lisk-db';
+import { TransactionPool } from '@liskhq/lisk-transaction-pool';
+import { dataStructures } from '@liskhq/lisk-utils';
+import { LiskValidationError, validator } from '@liskhq/lisk-validator';
+import { Logger } from '../../logger';
+import { Generator } from '../../types';
+import {
+	APIContext,
+	EventQueue,
+	StateMachine,
+	TransactionContext,
+	VerifyStatus,
+} from '../state_machine';
+import { Broadcaster } from './broadcaster';
+import { InvalidTransactionError } from './errors';
+import { GeneratorStore } from './generator_store';
+import {
+	PostTransactionRequest,
+	postTransactionRequestSchema,
+	PostTransactionResponse,
+	UpdateForgingStatusRequest,
+	updateForgingStatusRequestSchema,
+	UpdateForgingStatusResponse,
+} from './schemas';
+import { Consensus, Keypair } from './types';
+
+interface EndpointArgs {
+	keypair: dataStructures.BufferMap<Keypair>;
+	generators: Generator[];
+	consensus: Consensus;
+	chain: Chain;
+	stateMachine: StateMachine;
+	pool: TransactionPool;
+	broadcaster: Broadcaster;
+}
+
+interface EndpointInit {
+	logger: Logger;
+	generatorDB: KVStore;
+	blockchainDB: KVStore;
+}
+
+interface EndpointContext {
+	params: Record<string, unknown>;
+	logger: Logger;
+}
+
+export class Endpoint {
+	private readonly _keypairs: dataStructures.BufferMap<Keypair>;
+	private readonly _generators: Generator[];
+	private readonly _consensus: Consensus;
+	private readonly _stateMachine: StateMachine;
+	private readonly _pool: TransactionPool;
+	private readonly _broadcaster: Broadcaster;
+	private readonly _chain: Chain;
+
+	private _logger!: Logger;
+	private _generatorDB!: KVStore;
+	private _blockchainDB!: KVStore;
+
+	public constructor(args: EndpointArgs) {
+		this._keypairs = args.keypair;
+		this._generators = args.generators;
+		this._consensus = args.consensus;
+		this._chain = args.chain;
+		this._stateMachine = args.stateMachine;
+		this._pool = args.pool;
+		this._broadcaster = args.broadcaster;
+	}
+
+	public init(args: EndpointInit) {
+		this._logger = args.logger;
+		this._generatorDB = args.generatorDB;
+		this._blockchainDB = args.blockchainDB;
+	}
+
+	public async postTransaction(ctx: EndpointContext): Promise<PostTransactionResponse> {
```

Comment:
This function is copied from transport, and input is updated to `EndpointContext`